freqtrade/exchange/bitget.py

```diff
@@ -0,0 +1,249 @@
+import logging
+from datetime import datetime, timedelta, timezone
+
+import ccxt
+
+from freqtrade.constants import BuySell
+from freqtrade.enums import CandleType, MarginMode, PriceType, TradingMode
+from freqtrade.exceptions import DDosProtection, ExchangeError, OperationalException, TemporaryError
+from freqtrade.exchange import Exchange
+from freqtrade.exchange.common import retrier
+from freqtrade.exchange.exchange_types import FtHas, OHLCVResponse
+from freqtrade.util.datetime_helpers import dt_now, dt_ts
+
+
+logger = logging.getLogger(__name__)
+
+
+class Bitget(Exchange):
+    """
+    Bitget exchange class. Contains adjustments needed for Freqtrade to work
+    with this exchange.
+
+    Please note that this exchange is not included in the list of exchanges
+    officially supported by the Freqtrade development team. So some features
+    may still not work as expected.
+    """
+
+    _ft_has: FtHas = {
+        "ohlcv_candle_limit": 1000,
+        "ohlcv_has_history": True,
+        "order_time_in_force": ["GTC", "FOK", "IOC"],
+        "trades_has_history": True,
+    }
+    _ft_has_futures: FtHas = {
+        "ohlcv_has_history": True,
+        "mark_ohlcv_timeframe": "4h",
+        "funding_fee_timeframe": "8h",
+        "stoploss_on_exchange": True,
+        "stoploss_order_types": {"limit": "limit", "market": "market"},
+        "stop_price_prop": "stopPrice",
+        "stop_price_type_field": "triggerBy",
+        "stop_price_type_value_mapping": {
+            PriceType.LAST: "last_price",
+            PriceType.MARK: "mark_price",
+            PriceType.INDEX: "index_price",
+        },
+    }
+
+    _supported_trading_mode_margin_pairs: list[tuple[TradingMode, MarginMode]] = [
+        (TradingMode.FUTURES, MarginMode.ISOLATED)
+    ]
+
+    @property
+    def _ccxt_config(self) -> dict:
+        config = {}
+        if self.trading_mode == TradingMode.SPOT:
+            config.update({"options": {"defaultType": "spot"}})
+        elif self.trading_mode == TradingMode.FUTURES:
+            config.update({"options": {"defaultType": "swap"}})
+        config.update(super()._ccxt_config)
+        return config
```

Comment:
Is this really necessary?
this is the default, really.
Overriding for spot is only necessary if spot is not the default in ccxt - which i don't think is the case for bitget.

Reply:
get it, i will remove it